File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> nums3(n+m);
        if (m == 0)
            nums1 = nums2;
        else 
        {
            int i = 0;
            int j = 0;
            int k = 0;
            while (i < m && j < n)
            {
                if (nums1[i] <= nums2[j])
                    nums3[k++] = nums1[i++];

                else nums3[k++] = nums2[j++];
            }
            while (i < m) { nums3[k++] = nums1[i++]; }
            while (j < n) { nums3[k++] = nums2[j++]; }
            nums1 = nums3;
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        nums3 = [0]*(m+n)
        if m==0:
            nums1[:] = nums2
        else:
            i,j,k=0,0,0
            while i<m and j<n:
                if nums1[i] < nums2[j]:
                    nums3[k] = nums1[i]
                    i+=1
                else:
                    nums3[k]= nums2[j]
                    j+=1
                k+=1
            while i < m:
                nums3[k] = nums1[i]
                i += 1
                k += 1
            while j<n:
                nums3[k]=nums2[j]
                j+=1
                k+=1
            nums1[:]= nums3